### PR TITLE
implement vectorized evaluation for builtinLeftBinarySig

### PR DESCRIPTION
**Pull request by icditwang**
### What problem does this PR solve? 
implement vectorized evaluation for builtinLeftBinarySig

### What is changed and how it works?
`BenchmarkVectorizedBuiltinStringFunc/builtinLeftBinarySig-VecBuiltinFunc-8         	  100000	     13037 ns/op	       0 B/op	       0 allocs/op 

BenchmarkVectorizedBuiltinStringFunc/builtinLeftBinarySig-NonVecBuiltinFunc-8      	   50000	     39549 ns/op	       0 B/op	       0 allocs/op`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_json_vec.go [hunk 1]
```diff
@@ -581,4 +581,4 @@ func (b *builtinJSONUnquoteSig) vecEvalString(input *chunk.Chunk, result *chunk.
 		result.AppendString(res)
 	}
 	return nil
-}
+}
```
Comment (wshwsh12) [inline expression/builtin_json_vec.go]: change the code by accident?

File: expression/builtin_string_vec.go [hunk 2]
```diff
@@ -715,11 +715,42 @@ func (b *builtinFindInSetSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+		return err
+	}
+	defer b.bufAllocator.put(buf2)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf2); err != nil {
+		return err
+	}
+	left := buf2.Int64s()
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
```
Comment (wshwsh12) [inline expression/builtin_string_vec.go]: `buf.isnull() || buf2.isnull()`

**Comment by wshwsh12** [comment]
Please fix the CI fail.

File: expression/builtin_json_vec.go (hunk 1 above)
Comment (icditwang) [reply expression/builtin_json_vec.go]: I think it's my master branch‘s problem.

File: expression/builtin_string_vec.go [hunk 3]
```diff
@@ -715,11 +715,42 @@ func (b *builtinFindInSetSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+		return err
+	}
+	defer b.bufAllocator.put(buf2)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf2); err != nil {
+		return err
+	}
+	left := buf2.Int64s()
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) || buf2.IsNull(i) {
```
Comment (SunRunAway) [inline expression/builtin_string_vec.go]: we should call `result.AppendNull` here.

File: expression/builtin_string_vec_test.go [hunk 4]
```diff
@@ -174,6 +176,14 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
 	},
 	ast.Left: {
 		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString, types.ETInt}},
+		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString, types.ETInt},
+			childrenFieldTypes: []*types.FieldType{{Tp: mysql.TypeString, Flag: mysql.BinaryFlag, Collate: charset.CollationBin},
```
Comment (SunRunAway) [inline expression/builtin_string_vec_test.go]: Why do we need to specify `childrenFieldTypes` here?

Comment (icditwang) [reply expression/builtin_string_vec_test.go]: According to the test on my local machine,it didn't run this func if we dong't use it.

File: expression/builtin_string_vec.go (hunk 3 above)
Comment (icditwang) [reply expression/builtin_string_vec.go]: ok.

**Pull request merged**